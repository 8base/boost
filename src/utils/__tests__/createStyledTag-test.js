import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from '../../';
import { createStyledTag } from '../';

describe('As developer, I can create tag by provided theme and passed styles.', () => {
  const theme = {
    name: {
      modifiers: {
        kind: {
          primary: {
            color: '#000',
          },
          secondary: {
            color: '#fff',
          },
        },
        size: {
          xs: {
            width: '100px',
          },
          md: {
            width: '200px',
          },
        },
      },
      defaults: {
        kind: 'primary',
      },
    },
  };

  it('with style object', () => {
    const StyledTag = createStyledTag('name', {
      height: '10px',
    });

    const tree = renderer.create(<ThemeProvider theme={ theme }><StyledTag tagName="div" size="md">Children</StyledTag></ThemeProvider>);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('with style function', () => {
    const StyledTag = createStyledTag('name', (props) => ({
      height: '10px',
      gridGap: props.gap,
    }));

    const tree = renderer.create(<ThemeProvider theme={ theme }><StyledTag tagName="div" gap="10px">Children</StyledTag></ThemeProvider>);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
