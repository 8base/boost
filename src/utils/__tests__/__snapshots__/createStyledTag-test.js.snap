// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`As developer, I can create tag by provided theme and passed styles. with style function 1`] = `
.emotion-0 {
  height: 10px;
  grid-gap: 10px;
  color: #000;
}

<div
  className="emotion-0 emotion-1"
>
  Children
</div>
`;

exports[`As developer, I can create tag by provided theme and passed styles. with style object 1`] = `
.emotion-0 {
  height: 10px;
  color: #000;
  width: 200px;
}

<div
  className="emotion-0 emotion-1"
>
  Children
</div>
`;
